src/main/java/com/avispl/symphony/dal/avdevices/encoderdecoder/haivision/xdecoder/HaivisionXDecoderCommunicator.java

```diff
@@ -2643,24 +3128,25 @@ private void populateStreamConfigCaseTSOverSRT(Map<String, String> stats, Stream
 	 * @param advancedControllableProperties is the list that store all controllable properties
 	 * @param streamName name of stream
 	 * @param controllableProperty name of controllable property
-	 *
 	 * @throws ResourceNotReachableException when fail to control stream
 	 */
-	private void streamControl(Map<String, String> stats, List<AdvancedControllableProperty> advancedControllableProperties, String streamName,
-			String controllableProperty) {
+	private void streamControl(Map<String, String> stats, List<AdvancedControllableProperty> advancedControllableProperties, String streamName, String controllableProperty) {
 		StreamControllingMetric streamControllingMetric = StreamControllingMetric.getByName(controllableProperty);
 
-		Optional<StreamConfig> cachedStreamConfig = cachedStreamConfigs.stream().filter(config -> config.getName().equals(streamName) || config.getDefaultStreamName().equals(streamName)).findFirst();
-		if (!cachedStreamConfig.isPresent()) {
+		Optional<StreamConfig> cachedStreamConfig = cachedStreamConfigsFiltered.stream().filter(config -> config.getName().equals(streamName) || config.getDefaultStreamName().equals(streamName))
```

Comment:
the same here - getName() will always be non-null here, right?

Reply:
yes! getName() can't be null here. If the user does not set the name for the stream, the decoder API always returns the streamName = "(None)" by default.